test/common/stats/allocator_impl_test.cc

```diff
@@ -125,6 +133,166 @@ TEST_F(AllocatorImplTest, RefCountDecAllocRaceSynchronized) {
   EXPECT_FALSE(alloc_.isMutexLockedForTest());
 }
 
+TEST_F(AllocatorImplTest, ForEachCounter) {
```

Comment:
can you add another test where you supply nullptr for the lambdas to each foreach? There'll be not much to check; just want to avoid crashing.

Reply:
Done.